contracts/external/cw-bounties/src/contract.rs

```diff
@@ -0,0 +1,272 @@
+}
+
+pub fn pay_out(
+    deps: DepsMut,
+    env: Env,
+    id: u64,
+    recipient: String,
+) -> Result<Response, ContractError> {
+    // Check bounty exists
+    let mut bounty = BOUNTIES.load(deps.storage, id)?;
+
+    // Check bounty is open
+    if bounty.status != BountyStatus::Open {
+        return Err(ContractError::NotOpen {});
+    }
+
+    // Validate recipient address
+    deps.api.addr_validate(&recipient)?;
+
+    // Set bounty status to claimed
+    bounty.status = BountyStatus::Claimed {
+        claimed_by: recipient.clone(),
+        claimed_at: env.block.time.seconds(),
+    };
+    BOUNTIES.save(deps.storage, id, &bounty)?;
+
+    // Message to pay out remaining funds to recipient
+    let msg = BankMsg::Send {
+        to_address: recipient.clone(),
+        amount: vec![bounty.clone().amount],
+    };
+
+    Ok(Response::new()
+        .add_message(msg)
+        .add_attribute("action", "pay_out_bounty")
+        .add_attribute("bounty_id", id.to_string())
+        .add_attribute("amount", bounty.amount.to_string())
+        .add_attribute("recipient", recipient))
+}
+
+pub fn update(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    id: u64,
+    new_amount: Coin,
+    title: String,
+    description: Option<String>,
+) -> Result<Response, ContractError> {
+    // Check bounty exists
+    let bounty = BOUNTIES.load(deps.storage, id)?;
+
+    // Check bounty is open
+    if bounty.status != BountyStatus::Open {
+        return Err(ContractError::NotOpen {});
+    }
+
+    // Update bounty
+    BOUNTIES.save(
+        deps.storage,
+        bounty.id,
+        &Bounty {
+            id: bounty.id,
+            amount: new_amount.clone(),
+            title,
+            description,
+            status: bounty.status,
+            created_at: bounty.created_at,
+            updated_at: Some(env.block.time.seconds()),
+        },
+    )?;
+
+    // Check if amount is greater or less than original amount
+    let old_amount = bounty.amount;
+    let res = Response::new()
+        .add_attribute("action", "update_bounty")
+        .add_attribute("bounty_id", id.to_string())
+        .add_attribute("amount", new_amount.amount.to_string());
+
+    match new_amount.amount.cmp(&old_amount.amount) {
+        Ordering::Greater => {
+            // If new amount is greater, check funds sent plus
+            // original amount match new amount
+            let sent_amount = must_pay(&info, &new_amount.denom)?;
+            if sent_amount + old_amount.amount != new_amount.amount {
+                return Err(ContractError::InvalidAmount {
+                    expected: new_amount.amount - old_amount.amount,
+                    actual: sent_amount + old_amount.amount,
+                });
+            }
+            Ok(res)
+        }
+        Ordering::Less => {
+            // If new amount is less, pay out difference to owner
+            let diff = old_amount.amount - new_amount.amount;
```

Comment:
Also here, denoms might be different between old and new amount. Would be also great allowing updating Bounty with different denom. In this case old amount is send back and only new amount needs to be checked.

Reply:
:clap:

Reply:
Found some other stuff. If u don't mind, will make some changes with tests and create PR.

Reply:
check PR here: https://github.com/DA0-DA0/dao-contracts/pull/720